### PR TITLE
Codify how to remove TreeArtifact base directories in `pkg_files` 

### LINKED ISSUE #269: Packaging `strip_prefix` does not function on or within TreeArtifacts (directories)
`strip_prefix` for rules within rules_pkg removes directory components from paths.  This is very useful for users who wish to maintain parts of directory structures found in their source trees.

A potential misconception, though, is to what it applies.  It only applies to directories that can be known at analysis time.  TreeArtifacts and directories within do not seem to be considered.

Examples to demonstrate this are forthcoming.

This can be resolved by having the packaging rules do this at runtime, or via documentation that explains how to do prefix stripping for such artifacts.

### PR DESCRIPTION
**Pull request by nacl**
We often have TreeArtifacts (directory artifacts) that are built to be installed
directly onto the filesystem in packages, but their names are often not correct.
They, thus must be renamed.  There are two ways to do this after we submit #319.

1) Rename the TreeArtifact to the destination
2) Rename the TreeArtifact to nothing

I believe that (2) is clearer, as the `prefix` provided to `pkg_files` stands
out more than an entry in `renames`.  This change codifies a workaround we have
tried out internally -- renaming to "nothing".  "Nothing" in this case, is represented by the
constant `REMOVE_BASE_DIRECTORY`, (1) is still allowed.

This change does not attempt to do anything related to updating paths inside
TreeArtifacts.  This will be attempted in a follow-on change.

Tests and documentation updates are also provided.

This change also moves the common analysis test boilerplate to its own location
in `pkg/tests/util/defs.bzl`, where it will be more generally accessible in
future changes.

Partially resolves #269.

### CODE REVIEW

File: pkg/mappings.bzl [hunk 1]
```diff
@@ -24,14 +24,16 @@ the following:
...
-here.  TODO(nacl): implement one.
+here.
 """
 
 load("@bazel_skylib//lib:paths.bzl", "paths")
 load("//:providers.bzl", "PackageDirsInfo", "PackageFilegroupInfo", "PackageFilesInfo", "PackageSymlinkInfo")
 
 _PKGFILEGROUP_STRIP_ALL = "."
 
+REMOVE_BASE_DIRECTORY = ""
```
Comment (aiuto) [inline pkg/mappings.bzl:35]: Does this all still work if we made this an impossible value, like  chr(0)

**Review by aiuto — APPROVED** [review]
I'm sorry I missed this one.

File: pkg/mappings.bzl (hunk 1 above)
Comment (nacl) [reply pkg/mappings.bzl:35]: Not without modifications elsewhere.  Let me see how much it would take.

**Comment by nacl** [comment]
> I'm sorry I missed this one.

No worries, thanks for taking a look!

File: pkg/mappings.bzl (hunk 1 above)
Comment (nacl) [reply pkg/mappings.bzl]: It turned out to be pretty easy.  PTAL.

**Review by aiuto — APPROVED** [review]

**Pull request merged**